packages/liquidator/src/proxyTransactionWrapper.js

```diff
@@ -0,0 +1,288 @@
+            liquidatorReserveCurrencyAddress: 0x123... // address of the reserve currency for the bot to trade against
+            maxReserverTokenSpent: "10000" // define the maximum amount of reserve currency the bot should use in 1tx. }
+   * */
+  constructor({
+    web3,
+    financialContract,
+    gasEstimator,
+    syntheticToken,
+    collateralToken,
+    account,
+    dsProxyManager = undefined,
+    isUsingDsProxyToLiquidate = false,
+    proxyTransactionWrapperConfig
+  }) {
+    this.web3 = web3;
+    this.financialContract = financialContract;
+    this.gasEstimator = gasEstimator;
+    this.syntheticToken = syntheticToken;
+    this.collateralToken = collateralToken;
+    this.account = account;
+    this.dsProxyManager = dsProxyManager;
+
+    // Helper functions from web3.
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+
+    this.isUsingDsProxyToLiquidate = isUsingDsProxyToLiquidate;
+
+    // TODO: refactor the router to pull from a constant file.
+    const defaultConfig = {
+      isUsingDsProxyToLiquidate: {
+        value: false,
+        isValid: x => {
+          return typeof x == "boolean";
+        }
+      },
+      uniswapRouterAddress: {
+        value: "0x7a250d5630b4cf539739df2c5dacb4c659f2488d",
+        isValid: x => {
+          return this.web3.utils.isAddress(x);
+        }
+      },
+      uniswapFactoryAddress: {
+        value: "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f",
+        isValid: x => {
+          return this.web3.utils.isAddress(x);
+        }
+      },
+      liquidatorReserveCurrencyAddress: {
+        value: "",
+        isValid: x => {
+          return this.web3.utils.isAddress(x) || x === "";
+        }
+      },
+      maxReserverTokenSpent: {
+        value: MAX_UINT_VAL,
+        isValid: x => {
+          return typeof x == "string";
+        }
+      }
+    };
+
+    // Validate and set config settings to class state.
+    const configWithDefaults = createObjectFromDefaultProps(proxyTransactionWrapperConfig, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // Preform some basic initalization sanity checks.
+    if (this.isUsingDsProxyToLiquidate) {
+      assert(
+        this.dsProxyManager && this.dsProxyManager.getDSProxyAddress(),
+        "DSProxy Manger has not yet been initialized!"
+      );
+      assert(this.dsProxyManager != undefined, "Cant use dsProxy to liquidate if the client is set to undefined!");
+      assert(
+        this.web3.utils.isAddress(this.liquidatorReserveCurrencyAddress),
+        "Must provide a reserve currency address to use the proxy transaction wrapper!"
+      );
+    }
+
+    this.reserveToken = new web3.eth.Contract(getAbi("ExpandedERC20"), this.liquidatorReserveCurrencyAddress);
+    this.ReserveCurrencyLiquidator = getTruffleContract("ReserveCurrencyLiquidator", web3, "latest");
+  }
+
+  // TODO: wrap this into a common util.
```

Comment:
This needs to be refactored. I did not put it in this PR as it is in a few other places. I added an issue to track this task here: https://github.com/UMAprotocol/protocol/issues/2789